Adds extra event types

sthulb opened the pull request:
This adds support for CloudWatch Events, allowing for:
- AutoScaling
- Console Sign-In
- EC2
- Schedule

Commit:
Adds extra event types

This adds support for CloudWatch Events, allowing for:
- AutoScaling
- Console Sign-In
- EC2
- Schedule

tj commented on cloudwatch/cloudwatch.go:
```diff
@@ -0,0 +1,216 @@
+type Detail interface {
+	List() []string
+	Get(key string) (interface{}, bool)
+}
```
I wonder if instead of getting too opinionated here with interfaces, if we should just leave Details as `json.RawMessage` and provide the other struct(s), making the user unmarshal that part. 

I haven't looked at all the possible events but I think that might be a bit safer. Let me know what you think!

sthulb replied:
That could work! Makes the switch go away.

Commit:
Remove detail interface

Instead, we store the detail as json.RawMessage and expect the user to
unmarshal it

Commit:
Make ConsoleSignInDetail generic for API & Console events

Signed-off-by: Simon Thulbourn <simon+github@thulbourn.com>

sthulb commented:
@tj Hey! The changes have been made, can we merge?

tj commented on cloudwatch/data/cloudwatch_event_ec2_launch.json:
```diff
@@ -0,0 +1,26 @@
+  "time": "2015-11-11T21:31:47Z",
+  "region": "us-east-1",
+  "resources": [
+    "arn:aws:autoscaling:us-east-1:123456789012:autoScalingGroup:eb56d16b-bbf0-401d-b893-d5978ed4a025:autoScalingGroupName/sampleLuanchSucASG",
```
do we need this file? not seeing it referenced anywhere

tj commented on cloudwatch/cloudwatch.go:
```diff
@@ -0,0 +1,124 @@
+
+}
+
+func keyValue(i interface{}, key string) (interface{}, bool) {
```
keyValue and listKey are deadcode now with the recent change I think

tj commented:
looks good to me other than the stuff that could be removed!

Commit:
Remove old things

tj commented:
Awesome :D thankya!

sthulb commented:
@tj You're welcome! I'd rather have this in one place :)
